libkineto/src/CuptiActivityProfiler.cpp

```diff
@@ -288,7 +289,10 @@ void CuptiActivityProfiler::processTraceInternal(ActivityLogger& logger) {
 
   for (const auto& session : sessions_) {
     LOG(INFO) << "Processing child profiler trace";
-    session->processTrace(logger);
+    session->processTrace(logger,
+        std::bind(&CuptiActivityProfiler::cpuActivity, this, std::placeholders::_1),
+        captureWindowStartTime_,
+        captureWindowEndTime_);
```

Comment:
Why are we passing these 3 new arguments? could you please add some comments or description in the commit, I don't see it used, other than a new virtual function in IActivityProfiler.

Reply:
I have implemented my kineto plugin in an externel repo, where i used this new virtual function. 

According to https://github.com/pytorch/kineto/pull/662#issuecomment-1298935447,  I implemented my ActivityProfiler and ActivityProfilerSession.  My ActivityProfiler catching events for kernels, runtime apis, correlation_external_ids and so on. Like cupti, We also only save events catched in the range from captureWindowStartTime_ to captureWindowEndTime_.  So, when `processTrace`, we need to know this two timepoint.  And we want to link our runtime apis and kernels  to  their related cpu op. So,  we passing `CuptiActivityProfiler::cpuActivity` function.

Sadly for that our repo is not OSS yet, It's ongoing. So, I can't share a link for my code.

Reply:
yea @fwenguang  just add a comment that cpuActivity() function here is getting the linked cpuActivity :)
appreciate you implementing this!